lib/commons/dom/is-visible.js

```diff
@@ -13,14 +13,187 @@ const clipRegex = /rect\s*\(([0-9]+)px,?\s*([0-9]+)px,?\s*([0-9]+)px,?\s*([0-9]+
 const clipPathRegex = /(\w+)\((\d+)/;
 
 /**
- * Determines if an element is hidden with a clip or clip-path technique
- * @method isClipped
+ * Determine whether an element is visible
+ * @method isVisible
  * @memberof axe.commons.dom
- * @private
- * @param  {CSSStyleDeclaration} style Computed style
- * @return {Boolean}
+ * @instance
+ * @param {HTMLElement} el The HTMLElement
+ * @param {Boolean} screenReader When provided, will evaluate visibility from the perspective of a screen reader
+ * @param {Boolean} recursed
+ * @return {Boolean} The element's visibilty status
+ */
+export default function isVisible(el, screenReader, recursed) {
+  if (!el) {
+    throw new TypeError(
+      'Cannot determine if element is visible for non-DOM nodes'
+    );
+  }
+
```

Comment:
I think this is better done with assert.
```suggestion
  assert(el, 'Cannot determine if element is visible for non-DOM nodes');
```